src/v/redpanda/admin/field_registry.h

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "absl/time/time.h"
+#include "serde/protobuf/base.h"
+#include "utils/named_type.h"
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace admin {
+
+/// Generic reusable accessor for looking up the value of a specific field on an
+/// object
+template<typename T>
+struct field_accessor_info {
+    // clang-format off
+    using int64_getter = named_type<std::function<int64_t(T&)>, struct int64_getter_tag>;
```

Comment:
should these be std::function or ss::noncopyable_function or even absl::FunctionRef?